enterprise_gateway/services/external/k8s_client.py

```diff
@@ -0,0 +1,25 @@
+"""Contains factory to create kubernetes api client instances using a single confguration"""
+import os
+
+from kubernetes import client, config
+
+
+class KubernetesClientFactory:
+    """Manages kubernetes client creation from environment variables"""
+    def __init__(self):
```

Comment:
Nit: It is a good idea to forward-declare instance variables in Python. This helps your IDE/Typechecker know about the existence of this instance variable.  Underscores also hint that these are private.

```suggestion

class KubernetesClientFactory:
    """Manages kubernetes client creation from environment variables"""
    
    _kubernetes_config : client.Configuration
    
    def __init__(self):
```